google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/ManagedVmDeploymentRunConfigurationEditor.java

```diff
@@ -114,9 +140,56 @@ protected void resetEditorFrom(ManagedVmDeploymentConfiguration configuration) {
   @Override
   protected void applyEditorTo(ManagedVmDeploymentConfiguration configuration)
       throws ConfigurationException {
+    configuration.setUserSpecifiedArtifact(isUserSpecifiedPathDeploymentSource());
+    configuration.setUserSpecifiedArtifactPath(userSpecifiedArtifactFileSelector.getText());
     configuration.setDockerFilePath(dockerFilePathField.getText());
     configuration.setAppYamlPath(appYamlPathField.getText());
     configuration.setConfigType(getConfigType());
+
+    setDeploymentSourceName(configuration.getUserSpecifiedArtifactPath());
+    toggleJarWarSelector();
+    validateConfiguration();
+  }
+
+  private void toggleJarWarSelector() {
```

Comment:
I like your suggestion better actually. Updated the branch with both of your comments.